l10n_es_ticketbai/models/account_invoice.py

```diff
@@ -357,10 +317,27 @@ def tbai_is_invoice_refund(self):
         return res
 
     def tbai_get_value_serie_factura(self):
-        sequence = self.journal_id.invoice_sequence_id
+        if self.journal_id.refund_sequence \
+                and self.type in ['in_refund', 'out_refund']:
+            if self.env['ir.module.module'].search([
+                ('name', '=', 'l10n_es_account_invoice_sequence'),
+                ('state', '=', 'installed')
+            ]):
+                sequence = self.journal_id.refund_inv_sequence_id
+            else:
+                sequence = self.journal_id.refund_sequence_id
+        else:
+            if self.env['ir.module.module'].search([
+                ('name', '=', 'l10n_es_account_invoice_sequence'),
+                ('state', '=', 'installed')
+            ]):
+                sequence = self.journal_id.invoice_sequence_id
+            else:
+                sequence = self.journal_id.sequence_id
```

Comment:
@pedrobaeza El módulo account_invoice_sequence hace que el campo core de odoo del modelo account.journal sequence_id sea el que se utiliza para los asientos (cuando sin el módulo era el de las facturas). El módulo añade el campo invoice_sequence_id para las facturas (hace lo mismo con las rectificativas).

De esta manera, cuando account_invoice_sequence no está instalado y quiero obtener la secuencia de las facturas, tengo que utilizar el campo sequence_id, pero si se encuentra instalado necesito invoice_sequence_id, si no voy a obtener la secuencia de los asientos.

Reply:
Pero eso es lo que no tiene sentido, que necesites acceder a la secuencia. Si tenéis ese acople, malo. ¿Por qué necesitáis eso?

Reply:
@pedrobaeza Al mandar el fichero TicketBAI a hacienda, tienen que ir separados la serie de la factura y el número de la factura. Por ejemplo, para la factura INV/2021/0005, hay que enviar por una parte INV/2021/ y por la otra 0005.

Sin acceder a la secuencia, no veo manera de realizar esta separación, ya que no se puede saber donde separar. En cambio, accediendo a la secuencia se puede obtener la longitud del prefijo y realizar la separación.

Reply:
Dentro de que este requisito es bastante absurdo por parte de la administración, ya que el número es todo, aún accediendo a la secuencia, tienes muchas variables asociadas a los prefijos, por lo que no creo que esa técnica sea la adecuada. La técnica adecuada, que creo que lo hacemos en `l10n_es_vat_book`, es recorrer la cadena empezando por atrás, y escaneas desde el primer carácter numérico encontrado, hasta el último carácter no numérico, y así ya puedes extrar el número sin depender de secuencias y su implementación.